internal/testrunner/runners/system/runner.go

```diff
@@ -747,6 +759,43 @@ func validateFields(docs []common.MapStr, fieldsValidator *fields.Validator, dat
 	return nil
 }
 
+func getDocEventCount(events map[string]interface{}) int {
+	for k, v := range events {
+		if reflect.ValueOf(v).Kind() == reflect.Slice {
+			e := events[k].([]interface{})
+			logger.Debugf("found %d events under \"%s\"", len(e), k)
+			return len(e)
+		}
+	}
+	return 0
+}
+
+func getEventCount(docs []common.MapStr) (lenEvents int, err error) {
+	for _, doc := range docs {
+		var message interface{}
+		var events interface{}
+
+		if message, err = doc.GetValue("message"); err != nil {
+			return 0, testrunner.ErrTestCaseFailed{
+				Reason:  fmt.Sprintf("could not find message in doc %v", doc),
+				Details: err.Error(),
```

Comment:
From what I've seen, the `httpjson` input packages the response received from the target api within a `message` object. If the response contains multiple events, these are packaged as an array under an inner object whose name is not mandatory, for example `"events": [{},{},{}]`, `"data": [{},{},{}]` etc.

The `func getEventCount()` is only called when the assertion regarding the number of those events was previously configured in the system test, so it checks for the existence of `message` before calling `getDocEventCount()` to look for the arbitrarily named array (`"events": []` or `"data": []` etc.) and add its length to the number of events. 

See also: https://github.com/elastic/beats/blob/main/x-pack/filebeat/input/httpjson/input.go#L272

Reply:
>> The hit count (length of the .hits.hits[] in the Elasaticsearch response)

I looked at that first & found at this point in the testing, `response.split:` has not been applied, so API calls to a source other than Elasticsearch will have an array of events packaged inside a single `"message"`. I checked for `httpjson` and there were no system test cases where the response was NDJSON, it's always JSON with an arbitrarily named top level element inside `"message"`. I also tried adding multiple elements into the response & that didn't work atall (expected).

Reply:
@jsoriano any message may contain one or more events. This logic is counting events within messages, so I think `event_count` remains accurate

@andrewkroh when I was looking at the `hits.hits` vs `hits.total` my thought was a future issue to add `hit_count` as an alternative configuration parameter, in parallel to `event_count`

That would allow assertions to cover both scenarios as appropriate for each integration and ensure the distinction between hits & events is explicit

Prior to this PR, the existing logic admits the possibility of multiple `docs` so the assertion is designed to handle this as/when such test-cases  are added in a possible future.

I also added `TestResult.EventCount` to track the sum across a request as a whole as it's theoretically possible for a test to have multiple hits, each containing a message where the message contains multiple events. Running both assertions for that edge case could require `event_count` to match the sum total of events across messages within all hits and the number of hits to match the expected total

Reply:
I'm not following something here.

Elastic Agent generates documents (aka events). Those documents are written to Elasticsearch. For a given system test I know the expected number of documents that should be in Elasticsearch. After the test completes I want to assert that there are exactly the expected amount of documents in Elasticsearch.

The contents of those documents are beyond the scope of this assertion. So there's no need to examine the `message`.

Reply:
@andrewkroh 

`Documents : Events` is a `1:n` relationship for at least one integration (`httpjson`)

 >> After the test completes I want to assert that there are exactly the expected amount of documents in Elasticsearch

This is easier to test (it's what I wrote first) - however, it was counting documents, not events. I can definitely do that.

I´ll add the the `hit_count` logic in parallel as an additional assertion.

Context: I've been working with the `httpjson` integration as my point of reference. In that integration, it appears `message` is split to constituent events downstream of the system test. So, `hits.total == len(hits.hits[]) == 1` every time.

Did you have a specific integration in mind when you conceived the assertion? I know others _do_ return `hits.total > 1` and I could use that as a test case to ensure both cases are covered & then look into the edge case where both assertions might w be defined. If not, I can just pick one.